src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -199,21 +218,70 @@ public DRSAccessURL getAccessUrlForObjectId(
 
     DrsId drsId = drsIdService.fromObjectId(objectId);
     Snapshot snapshot = snapshotService.retrieve(UUID.fromString(drsId.getSnapshotId()));
-    GoogleProjectResource projectResource = snapshot.getProjectResource();
 
-    DRSAccessMethod matchingAccessMethod =
-        getAccessMethodMatchingAccessId(accessId, drsObject)
-            .orElseThrow(
-                () ->
-                    new IllegalArgumentException(
-                        "No matching access ID " + accessId + " was found on object " + objectId));
+    BillingProfileModel billingProfileModel =
+        snapshot
+            .getFirstSnapshotSource()
+            .getDataset()
+            .getDatasetSummary()
+            .getDefaultBillingProfile();
+
+    Supplier<IllegalArgumentException> illegalArgumentExceptionSupplier =
+        () -> {
+          throw new IllegalArgumentException("No matching access ID was found for object");
+        };
+
+    CloudPlatformWrapper wrapper = CloudPlatformWrapper.of(billingProfileModel.getCloudPlatform());
+    if (wrapper.isGcp()) {
+      DRSAccessMethod matchingAccessMethod =
+          getAccessMethodMatchingAccessId(accessId, drsObject, TypeEnum.GS)
+              .orElseThrow(illegalArgumentExceptionSupplier);
+      return signGoogleUrl(snapshot, matchingAccessMethod);
+    } else if (wrapper.isAzure()) {
+      getAccessMethodMatchingAccessId(accessId, drsObject, TypeEnum.HTTPS)
```

Comment:
Just for my own understanding, why do we need to get the access method for Azure when it doesn't get used?

Reply:
Is this just to validate that this is a good access id?